sqlmesh/dbt/context.py

```diff
@@ -11,6 +11,7 @@
 from sqlmesh.dbt.manifest import ManifestHelper
 from sqlmesh.dbt.target import TargetConfig
 from sqlmesh.utils import AttributeDict
+from sqlmesh.utils.dbt import TARGET_TYPE_TO_DIALECT_OVERRIDE
```

Comment:
I'm not a fan of the override being done like this. IMO it should be a property in the target classes and we use that instead of type.